src/gateway/WebSocket.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-case-declarations */
+import WebSocket from 'ws';
+import fetch from 'node-fetch';
+import { GATEWAY_OPCODES, GATEWAY_CLOSE_CODES, EVENTS, Heartbeat, Identify, Payload } from './constants';
+import { client } from '../Client';
+
+export default async function Socket(token: string, intents: number) {

```

Comment:
While I have no issue with a functional approach, what is the general project philosophy?
The general rule in OOP is that you should do a function if a class is useless (if you have only a constructor and 1 method in said class for instance) but the socket connection is potentially more complex.

Again, as a temporary solution it's all good. And don't get me wrong there is nothing wrong with functions, I just felt like it was worth raising the question

Reply:
For the moment we are not following a philosophy, we are rather focused on establishing the base of the library and keeping things simple